### PR TITLE
Fix NullPointerException on empty sequence response

### PR DESCRIPTION
**Pull request by leblowl**
Hola,

Here is my attempt @ fixing http-kit/http-kit#247. I also added a test route to the test server for this example. I am not sure that was the best place to put it, if there is a unit testing section for this, then I can add something there. p.s. It was a little weird transitioning from Clojure to Java... count in clojure doesn't throw null pointers ! Also I have a question, where does this count method even come from, the body gets cast to an ISeq and that interface doesn't contain any count method from what I can tell. Many thanks

### CODE REVIEW

**Comment by leblowl** [comment]
Not sure what is up with the merge commit, if you have advice on how to prettify the history, I can do it...

**Comment by ptaoussanis** [comment]
No problem re: history, I'll sort that out.

Am wondering though: is there a particular reason we can't just return null for an empty seq here instead of still going through the 0-count DynamicBytes?

File: test/org/httpkit/server_test.clj [hunk 1]
```diff
@@ -110,7 +110,10 @@
                               :body "Hello World"}))
   (GET "/iseq" [] (fn [req] {:status 200
                             :headers {"Content-Type" "text/plain"}
```
Comment (venantius) [inline test/org/httpkit/server_test.clj]: could we fix the indentation on this `:headers` key? you've fixed the indentation for `:body` and now I can't ignore the `:headers` :stuck_out_tongue:

Comment (leblowl) [inline test/org/httpkit/server_test.clj]: thank you, can fix

**Comment by leblowl** [comment]
@ptaoussanis no reason, just wasn't aware I could. It works well & will be a bit quicker that way.

**Comment by leblowl** [comment]
^ something like this. not sure what line of code the comment is referring to here:  https://github.com/leblowl/http-kit/blob/a5cd1be385d4936dfea7afee6b17a5bff8dea519/src/java/org/httpkit/HttpUtils.java#L116... I can move it/reindent it if needs.

**Comment by ptaoussanis** [comment]
> not sure what line of code the comment is referring to here:

Looks like that may need to be one line down, for: `else if (body instanceof ByteBuffer) { return (ByteBuffer) body; }`

**Comment by ptaoussanis** [comment]
Thank you!

**Pull request merged**